dot/state/block.go

```diff
@@ -250,6 +251,39 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetHashesByNumber returns the block hashes with the given number
+func (bs *BlockState) GetHashesByNumber(num uint) ([]common.Hash, error) {
+	block, err := bs.GetBlockByNumber(num)
+	if err != nil {
+		return []common.Hash{}, fmt.Errorf("getting block by number %w", err)
+	}
+	blockHashes := bs.bt.GetAllBlocksAtNumber(block.Header.ParentHash)
+
+	hash, err := bs.bt.GetHashByNumber(num)
+	if err != nil {
+		if !errors.Is(err, blocktree.ErrNumLowerThanRoot) {
+			return blockHashes, fmt.Errorf("failed to get hash from blocktree: %w", err)
+		}
+
+		// if error is ErrNumLowerThanRoot, number has already been finalised, so check db
+		bh, err := bs.db.Get(headerHashKey(uint64(num)))
+		if err != nil {
+			return blockHashes, fmt.Errorf("cannot get block %d: %w", num, err)
+		}
+
+		if !slices.Contains(blockHashes, hash) {
+			blockHashes = append(blockHashes, common.NewHash(bh))
+		}
+
+		return blockHashes, nil
+	}
```

Comment:
```suggestion
	hash := block.Header.Hash()
```

Reply:
don't remember why i did that. must be some confusion, fixing it.